src/TinyUPnP.cpp

```diff
@@ -6,27 +6,29 @@
 #include "Arduino.h"
 #include "TinyUPnP.h"
 
+#ifdef UPNP_DEBUG
+#undef UPNP_DEBUG
+#define UPNP_DEBUG(...) Serial.print( __VA_ARGS__ )
+#define UPNP_DEBUGln(...) Serial.println( __VA_ARGS__ )
+#else
+#define UPNP_DEBUG(...)
+#define UPNP_DEBUGln(...)
+#endif
+
 IPAddress ipMulti(239, 255, 255, 250);  // multicast address for SSDP
 IPAddress connectivityTestIp(64, 233, 187, 99);  // Google
-char packetBuffer[UDP_TX_PACKET_MAX_SIZE];  // buffer to hold incoming packet UDP_TX_PACKET_MAX_SIZE=8192
```

Comment:
Will you agree to divide this PR into two:
1) anything that has to do with the debug macro + things I commented as OK
2) rest of the logic changes. Keep in mind that I am not sure I will allow the String usage, though. Moving from String to the arrays made my setup much more stable. And I intend to clean more String usages in the future.

Reply:
The Problem with NOT using Strings is that you either have to use Fixed Arrays or Dynamic Allocation.
Fixed Arrays like you did and that Consumes a lot of the data memory .. the size of the TinyUPnP Object now is 116 byte compared to about 10k before !!! that's a lot of wasted space for such a Tiny Library :D
the other alternative is to use dynamic allocation ..
and in that case the code will be more complex and you risk the possibility of memory leaks
the way I see it .. Strings are Great because of their Dynamic Nature and they never caused A problem with me before ..

Reply:
Most of the memory consumption was due to the compiled debug hard coded strings.
Let's start with that phase please so that you'll be convinced that it is true.
A 1000 byte array (UDP_TX_PACKET_MAX_SIZE) is better than using Strings which are in themselves dynamic allocation.
Anyway, please start with only the debug. This is a really good contribution. Let's consider the String after you'll see the memory consumption with just the debug as Macro.

Reply:
What package of TLS do you use so that I can experiment with it and find an optimum?